src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -424,8 +424,7 @@ private void processIter(DetailAST root, AstState astState) {
      */
     private static SortedSet<AbstractCheck> createNewCheckSortedSet() {
         return new TreeSet<>(
-                Comparator.<AbstractCheck, String>comparing(check -> check.getClass().getName())
-                        .thenComparing(AbstractCheck::getId,
+                Comparator.comparing(AbstractCheck::getId,
                                 Comparator.nullsLast(Comparator.naturalOrder()))
```

Comment:
https://pitest.org/quickstart/mutators/#EMPTY_RETURNS

We need to have different Checks in config. To let violations be sorted by check name.
But it might require non regular verify method, as we need strictly check order of violations, existing verify method might do comparison in non strict ways or by its own sorting.

Please look inside verify method on how it compares violations.

Reply:
One more mutation is required to be removed https://github.com/checkstyle/checkstyle/actions/runs/6723258174/job/18272949364?pr=13976#step:7:24